extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/JPAConfig.java

```diff
@@ -28,11 +29,11 @@ public class JPAConfig {
     private final Map<String, LazyPersistenceUnit> persistenceUnits;
 
     @Inject
-    public JPAConfig(JPAConfigSupport jpaConfigSupport) {
-
+    public JPAConfig() {
         Map<String, LazyPersistenceUnit> persistenceUnitsBuilder = new HashMap<>();
-        for (String persistenceUnitName : jpaConfigSupport.persistenceUnitNames) {
-            persistenceUnitsBuilder.put(persistenceUnitName, new LazyPersistenceUnit(persistenceUnitName));
+        for (PersistenceUnitDescriptor descriptor : PersistenceUnitsHolder.getPersistenceUnitDescriptors()) {
```

Comment:
I don't know about that race condition and I don't see any failing test, so I can't say...

What I can say:

* `JPAConfigSupport` [was introduced](https://github.com/quarkusio/quarkus/pull/11322/commits/51c87b827f8903bf6eb82b7c945917f4cfe5dd77#diff-dfa2066c93df2973e1785df934b4f2095919db1073791af001adcb1aa37b0d45) to "make JPAConfig less brittle"; not sure if it was about an actual bug or design concerns (best practices).
* `JPAConfigSupport` [initially contained more information than just persistence unit names](https://github.com/quarkusio/quarkus/pull/11322/commits/51c87b827f8903bf6eb82b7c945917f4cfe5dd77#diff-55c8984ab2a8fd0ffd9ecfb5e858c021a6b93f74d16b5222ec5809f40380eacb); that is no longer the case.
* `PersistenceUnitsHolder` is initialized at static init: 
  * https://github.com/quarkusio/quarkus/blob/aec43cd82087ef5f8d2059cb181e76bcb2801a5f/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java#L73-L79
* `JPAConfig` is supposedly only used at runtime; that makes sense since session factories are a runtime construct. Here are the only uses I found:
  * https://github.com/quarkusio/quarkus/blob/bc4955b027800abd07c066f63120bc37b532f634/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/metrics/HibernateMetricsRecorder.java#L24-L29
  * https://github.com/quarkusio/quarkus/blob/e9b4dcef561a207fb922437443b1557b2e113ade/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java#L98-L104
  * https://github.com/quarkusio/quarkus/blob/aec43cd82087ef5f8d2059cb181e76bcb2801a5f/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/HibernateOrmRecorder.java#L103-L105
  * https://github.com/quarkusio/quarkus/blob/1aec4e32af10cc90a54db1f104268694a2270043/extensions/hibernate-orm/runtime/src/main/java/io/quarkus/hibernate/orm/runtime/TransactionSessions.java#L18-L22
* Tests pass :shrug: 

If for some reason tests start failing because the constructor of `JPAConfig` ends up being executed during static init (and before `PersistenceUnitsHolder` is initialized), we can make `JPAConfig` a synthetic bean and force runtime init. Or, worst case, we can make `JPAConfig` `@ApplicationScoped`, so that it's created on first use.

In fact, I can create a quick PR to make `JPAConfig` a synthetic bean right now. It seems more correct, regardless of whether it's necessary, because that bean is fundamentally a runtime bean. I'll open that PR soon.